diff --git a/Dockerfile b/Dockerfile
index e19d958..e8250c6 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -190,19 +190,8 @@ RUN wget -O /linux.tar.xz "https://cdn.kernel.org/pub/linux/kernel/v${LINUX_VERS
 # verify
 	export GNUPGHOME="$(mktemp -d)"; \
 	for key in $LINUX_GPG_KEYS; do \
-		for mirror in \
-			ha.pool.sks-keyservers.net \
-			pgp.mit.edu \
-			hkp://p80.pool.sks-keyservers.net:80 \
-			ipv4.pool.sks-keyservers.net \
-			keyserver.ubuntu.com \
-			hkp://keyserver.ubuntu.com:80 \
-		; do \
-			if gpg --batch --verbose --keyserver "$mirror" --keyserver-options timeout=5 --recv-keys "$key"; then \
-				break; \
-			fi; \
-		done; \
-		gpg --batch --fingerprint "$key"; \
+		wget -O - "https://keyserver.ubuntu.com/pks/lookup?op=get&options=mr&search=0x$key" \
+			| gpg --batch --import || echo "note: key $key unavailable, skipping"; \
 	done; \
 	gpg --batch --verify /linux.tar.asc /linux.tar; \
 	gpgconf --kill all; \
